### PR TITLE
Add Developer Docs

### PR DESCRIPTION
**Pull request by micbar**
## Description
- Add Developer Docs build with hugo

## How to test
`make docs`
`cd hugo`
`hugo server -D`

Then check the result at `http://localhost:1313/`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by micbar** [comment]
@LukasHirt @PVince81 
I added a "skeleton" content in the `docs` folder.

Feel free to add more commits to this PR

File: .drone.star [hunk 1]
```diff
@@ -855,6 +855,94 @@ def buildRelease(ctx):
...
+						'refs/pull/**',
+						],
+					},
+				},
+			},
+			{
+				'name': 'downstream',
+				'image': 'plugins/downstream',
+				'settings': {
+					'server': 'https://cloud.drone.io/',
```
Comment (LukasHirt) [inline .drone.star]: We don't use our drone instance for this?

Comment (micbar) [reply .drone.star]: This is the central docs repo.

**Comment by micbar** [comment]
@LukasHirt Ah, you force pushed. I thought you would add another commit with your changes.

File: .drone.star [hunk 2]
```diff
@@ -855,6 +855,94 @@ def buildRelease(ctx):
 		}
 	]
 
+def website(ctx):
```
Comment (LukasHirt) [inline .drone.star]: What about calling this "docs" instead of "website"?

**Comment by micbar** [comment]
Now you reset, some of my changes.

File: .drone.star (hunk 2 above)
Comment (micbar) [reply .drone.star]: it is called `website` in all our repos. It publishes the content to `owncloud.github.io`

**Comment by LukasHirt** [comment]
> @LukasHirt Ah, you force pushed. I thought you would add another commit with your changes.

Was that wrong? I kept you still as the author and did just some smaller changes in the existing docs so thought it would be okay.

**Comment by LukasHirt** [comment]
> Now you reset, some of my changes.

Agh, I see. I'm sorry

**Comment by micbar** [comment]
Your branch was outdated, that is always the risk.

Let me figure this out with my local branch.

**Comment by LukasHirt** [comment]
If you now commit and then pull with autorebase does it overwrite your local changes?

**Comment by micbar** [comment]
@LukasHirt ok, re pushed

please add your stuff on top of it in a separate commit :-)

**Comment by LukasHirt** [comment]
@micbar Added a small commit with some small adjustments in the docs.

**Comment by dpakach** [comment]
@micbar, I have added a commit to fix the CI issue. Lets see if that changes anything. I'm confused why it's failing just on this PR. The drone config added here doesn't seems to affect the acceptance test pipeline.

**Pull request merged**